$NetBSD: patch-configure,v 1.7 2018/08/12 08:33:55 maya Exp $

Regen.

install .oct loadable modules with INSTALL_LIB to avoid stripping them

we pass and assume a given MACHINE_GNU_PLATFORM, but this package did not
use this value due to an error.
the guess it had for linux (x86_64-unknown-linux-gnu), seems to trample
over our own tools:
https://mail-index.netbsd.org/pkgsrc-users/2014/03/26/msg019464.html

Allow overriding MKOCTFILE_F77 so we can provide the full path to gfortran
(for building octave-forge packages like control)

Additional changes are due to autoconf changes.

--- configure.orig	2018-08-09 18:20:32.000000000 +0000
+++ configure
@@ -1733,6 +1733,7 @@ INCLUDE_NEXT_AS_FIRST_DIRECTIVE
 INCLUDE_NEXT
 LIB_CRYPTO
 pkglibexecdir
+runstatedir
 lispdir
 REPLACE_FDOPENDIR
 REPLACE_DIRFD
@@ -2092,6 +2093,7 @@ PYTHON
 DEFAULT_PAGER
 GNUPLOT
 GHOSTSCRIPT
+INSTALL_LIB
 LN_S
 REPRODUCIBLE_TAR_FLAGS
 TEXI2PDF
@@ -2221,7 +2223,6 @@ infodir
 docdir
 oldincludedir
 includedir
-runstatedir
 localstatedir
 sharedstatedir
 sysconfdir
@@ -2446,7 +2447,6 @@ datadir='${datarootdir}'
 sysconfdir='${prefix}/etc'
 sharedstatedir='${prefix}/com'
 localstatedir='${prefix}/var'
-runstatedir='${localstatedir}/run'
 includedir='${prefix}/include'
 oldincludedir='/usr/include'
 docdir='${datarootdir}/doc/${PACKAGE_TARNAME}'
@@ -2699,15 +2699,6 @@ do
   | -silent | --silent | --silen | --sile | --sil)
     silent=yes ;;
 
-  -runstatedir | --runstatedir | --runstatedi | --runstated \
-  | --runstate | --runstat | --runsta | --runst | --runs \
-  | --run | --ru | --r)
-    ac_prev=runstatedir ;;
-  -runstatedir=* | --runstatedir=* | --runstatedi=* | --runstated=* \
-  | --runstate=* | --runstat=* | --runsta=* | --runst=* | --runs=* \
-  | --run=* | --ru=* | --r=*)
-    runstatedir=$ac_optarg ;;
-
   -sbindir | --sbindir | --sbindi | --sbind | --sbin | --sbi | --sb)
     ac_prev=sbindir ;;
   -sbindir=* | --sbindir=* | --sbindi=* | --sbind=* | --sbin=* \
@@ -2845,7 +2836,7 @@ fi
 for ac_var in	exec_prefix prefix bindir sbindir libexecdir datarootdir \
 		datadir sysconfdir sharedstatedir localstatedir includedir \
 		oldincludedir docdir infodir htmldir dvidir pdfdir psdir \
-		libdir localedir mandir runstatedir
+		libdir localedir mandir
 do
   eval ac_val=\$$ac_var
   # Remove trailing slashes.
@@ -2998,7 +2989,6 @@ Fine tuning of the installation director
   --sysconfdir=DIR        read-only single-machine data [PREFIX/etc]
   --sharedstatedir=DIR    modifiable architecture-independent data [PREFIX/com]
   --localstatedir=DIR     modifiable single-machine data [PREFIX/var]
-  --runstatedir=DIR       modifiable per-process data [LOCALSTATEDIR/run]
   --libdir=DIR            object code libraries [EPREFIX/lib]
   --includedir=DIR        C header files [PREFIX/include]
   --oldincludedir=DIR     C header files for non-gcc [/usr/include]
@@ -9414,7 +9404,10 @@ fi
 
 
 
-INSTALL_SCRIPT="${INSTALL}"
+test -z "$INSTALL_SCRIPT" && INSTALL_SCRIPT='${INSTALL}'
+
+test -z "$INSTALL_LIB" && INSTALL_LIB='${INSTALL_SCRIPT}'
+
 
 
 ## Programs used when running Octave.
@@ -12974,6 +12967,11 @@ fi
 	  done
 	  ;;
 	-[lLR]*)
+	    case $ac_arg in
+	    -R*)
+	        ac_arg="-Wl,$ac_arg"
+	        ;;
+	    esac
 	    ac_exists=false
   for ac_i in $ac_cv_f77_libs; do
     if test x"$ac_arg" = x"$ac_i"; then
@@ -34971,13 +34969,13 @@ if test "$cross_compiling" = yes && test
   MKOCTFILE_LD_CXX=`echo "$LD_CXX" | $SED "s,$CROSS_TOOL_PREFIX,,"`
   MKOCTFILE_RANLIB=`echo "$RANLIB" | $SED "s,$CROSS_TOOL_PREFIX,,"`
 else
-  MKOCTFILE_AR="$AR"
-  MKOCTFILE_CC="$CC"
-  MKOCTFILE_CXX="$CXX"
-  MKOCTFILE_DL_LD="$DL_LD"
-  MKOCTFILE_F77="$F77"
-  MKOCTFILE_LD_CXX="$LD_CXX"
-  MKOCTFILE_RANLIB="$RANLIB"
+  MKOCTFILE_AR="${MKOCTFILE_AR-$AR}"
+  MKOCTFILE_CC="${MKOCTFILE_CC-$CC}"
+  MKOCTFILE_CXX="${MKOCTFILE_CXX-$CXX}"
+  MKOCTFILE_DL_LD="${MKOCTFILE_DL_LD-$DL_LD}"
+  MKOCTFILE_F77="${MKOCTFILE_F77-$F77}"
+  MKOCTFILE_LD_CXX="${MKOCTFILE_LD_CXX-$LD_CXX}"
+  MKOCTFILE_RANLIB="${MKOCTFILE_RANLIB-$RANLIB}"
 fi
 { $as_echo "$as_me:${as_lineno-$LINENO}: defining CROSS_TOOL_PREFIX to be $CROSS_TOOL_PREFIX" >&5
 $as_echo "$as_me: defining CROSS_TOOL_PREFIX to be $CROSS_TOOL_PREFIX" >&6;}
@@ -36091,6 +36089,8 @@ else
 # ifdef _MSC_VER
 #  include <malloc.h>
 #  define alloca _alloca
+# elif defined(__NetBSD__) || defined(__FreeBSD__) || defined(__DragonFly__) || defined(__OpenBSD__)
+#   include <stdlib.h>
 # else
 #  ifdef HAVE_ALLOCA_H
 #   include <alloca.h>
